server/controllers/gyms/SerachGymByName.ts

```diff
@@ -0,0 +1,31 @@
+/* eslint-disable no-unused-vars */
+import { NextFunction, Response, Request } from 'express';
+import Sequelize from 'sequelize';
+import { Gym, Image, Review } from '../../database/models';
+import CustomError from '../../utils';
+
+export default async function searchGymByName(req: Request, res: Response, next: NextFunction) {
+  try {
+    const gyms = await Gym.findAll({
+      where: {
+        gym_name: {
+          [Sequelize.Op.like]: `%${req.query.q}%`,
+        },
+      },

```

Comment:
try to exclude passwords from gym data,
And exclude other fields you don't need,
Also, I guess we need a Name for the user who owns the review